### ISSUE TITLE
Bug: Interface-based Abstractions Throws Missing FieldOrder Attribute

### ISSUE THREAD

**Issue by Texnomic** [jefffhaynes/BinarySerializer#140]
When using Interface-based Abstractions, The following case works **only if** Interface `IDomain` members are marked with `FieldOrder` Attributes. Otherwise `ThrowIfUnordered()` throws `InvalidOperationException` because it checked Interface `IDomain` members not type `Domain` members.

```csharp
public interface IDomain
{
        List<ILabel> Labels { get; }
        string Name { get; }
}

public class Domain : IBinarySerializable, IDomain
{
        [Ignore]
        public List<ILabel> Labels { get; }

        [Ignore]
        public string Name => ToString();

        public void Serialize(Stream Stream, Endianness Endianness, BinarySerializationContext Context)
        {
           //Code
        }

        public void Deserialize(Stream Stream, Endianness Endianness, BinarySerializationContext Context)
        {
            //Code
        }

}

public class Question : IQuestion
{
        [FieldOrder(0)]
        [SubtypeDefault(typeof(Domain))] //Check Issue #139 
        public IDomain Domain { get; set; }
}
```

**Comment by A016699** [comment]
Hi, I'm just run in the same problem. Do you think there will be fix soon ?

**Comment by jefffhaynes** [comment]
I'll work on this but it is a tricky problem.  The possibility is there to create an entire inheritance hierarchy of serializer attributes, which isn't really something that is currently supported.  Ignoring attributes in a derived (or implemented) class is actually just a special case of that hierarchy.  That being said, I certainly see the value.

**Comment by A016699** [comment]
I understand the problem, Just a little hint. It seems to work if I I use the SubtypeAttribute to - could be there a workaround for it ?

**Comment by jefffhaynes** [comment]
Fair point.  Maybe I'm thinking about it wrong.  Let me try a different approach.

**Comment by jefffhaynes** [comment]
Fixed in 8.5

**Comment by A016699** [comment]
Thank you, for your fast response, it is create lib and I love working with it.

**Comment by jefffhaynes** [comment]
Thanks!  Glad you like it!

**Comment by jefffhaynes** [comment]
Sorry, I have no idea what happened but apparently this exact issue isn't quite fixed yet.  Give me a few...

**Comment by jefffhaynes** [comment]
Ok, deserialization wasn't working.  Fixed in 8.5.1
